src/main/java/core/basesyntax/Car.java

```diff
@@ -1,5 +1,115 @@
-package core.basesyntax;
-
-public class Car {
-
-}
+package core.basesyntax;
+
+public class Car implements Cloneable {
+    private String name;
+    private String color;
+    private int weight;
+    private int maxSpeed;
+    private int seats;
+    private Engine engine;
+
+    public Car(String name, String color, int weight, int maxSpeed, int seats) {
+        this.name = name;
+        this.color = color;
+        this.weight = weight;
+        this.maxSpeed = maxSpeed;
+        this.seats = seats;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    public void setColor(String color) {
+        this.color = color;
+    }
+
+    public int getWeight() {
+        return weight;
+    }
+
+    public void setWeight(int weight) {
+        this.weight = weight;
+    }
+
+    public int getMaxSpeed() {
+        return maxSpeed;
+    }
+
+    public void setMaxSpeed(int maxSpeed) {
+        this.maxSpeed = maxSpeed;
+    }
+
+    public int getSeats() {
+        return seats;
+    }
+
+    public void setSeats(int seats) {
+        this.seats = seats;
+    }
+
+    public void setEngine(Engine engine) {
+    }
+
+    public void setEngine(int cilindersNumber, int volumeCilinders, int weightEngine,
+                          String engineType, String kppType) {
+        this.engine = new Engine(cilindersNumber,
+                volumeCilinders,
+                weightEngine,
+                engineType,
+                kppType);
+    }
+
+    public void startEngine() {
+        if (engine != null) {
+            engine.startEngine();
+        } else {
+            System.out.println("Engine doesn't exist!");
+        }
+    }
+
+    public void stopEngine() {
+        if (engine != null) {
+            engine.stopEngine();
+        } else {
+            System.out.println("Engine doesn't exist!");
+        }
+    }
+
+    @Override
+    public Car clone() throws CloneNotSupportedException {
+        Car clonedCar = new Car(
+                getName(),
+                getColor(),
+                getWeight(),
+                getMaxSpeed(),
+                getSeats());

```

Comment:
you can use private fields in the same class
```suggestion
        Car clonedCar = new Car(name, color, weight, maxSpeed, seats);
```

Reply:
Sorry, fixed :(